### PR TITLE
update to RDKit 2022.09.01

### PR DESCRIPTION
**Pull request by eloyfelix**

### CODE REVIEW

**Comment by eloyfelix** [comment]
sorry for bothering you @giordano... any clue why this might be happening?

- test pass in MacOs

- partially pass in Ubuntu but segfaults
```
     Testing Running tests...
Test Summary: | Pass  Total
io            |   12     12
Test Summary: | Pass  Total
drawing       |    3      3
Test Summary: | Pass  Total
calculators   |   16     16
Test Summary:   | Pass  Total
standardization |   10     10

signal (11): Segmentation fault
in expression starting at /home/runner/work/RDKitMinimalLib.jl/RDKitMinimalLib.jl/test/runtests.jl:153
set_2d_coords_aligned at /home/runner/.julia/artifacts/3921da617a6dfaead03ef2092bde61f9ebcf9c34/lib/librdkitcffi.so (unknown line)
set_2d_coords_aligned at /home/runner/work/RDKitMinimalLib.jl/RDKitMinimalLib.jl/src/coordinates.jl:39
set_2d_coords_aligned at /home/runner/work/RDKitMinimalLib.jl/RDKitMinimalLib.jl/src/coordinates.jl:38
unknown function (ip: 0x7fc5b4978693)
```

- Doesn't seem to find any function in windows (I just added it to the tests to see again what was going on)

```
     Testing Running tests...
io: Error During Test at D:\a\RDKitMinimalLib.jl\RDKitMinimalLib.jl\test\runtests.jl:37
  Got exception outside of a @test
  could not load symbol "get_mol":
  The specified procedure could not be found. 
```

**Comment by giordano** [comment]
Bug in the library?

**Comment by eloyfelix** [comment]
RDKit? this would be my best guess but I'm not an expert building stuff :P

**Comment by giordano** [comment]
Did you see https://github.com/rdkit/rdkit/pull/5598? It mentions some major changes to `set_2d_coords_aligned`, that's perhaps relevant?

**Comment by giordano** [comment]
> Doesn't seem to find any function in windows (I just added it to the tests to see again what was going on)

I already said elsewhere that I don't know what's the problem. [As far as I can tell](https://github.com/JuliaPackaging/Yggdrasil/pull/4715#issuecomment-1280007541)  symbols _are_ exported, I have no idea what else could be the problem and no way to debug anything: I'm not familiar with the operating system nor have access to it, in the unlikely case I was willing to look into the issue. Someone _else_ more familiar with windows will have to dig into this.

**Comment by eloyfelix** [comment]
I saw but seeing MacOs tests pass was extremely confusing to me. I'll investigate a bit more... thanks!

**Comment by giordano** [comment]
The only issue I can think of on Windows is that the dynamic loader can't resolve some ***other*** symbols in `librdkitcffi.dll` and prints out a completely unhelpful and misleading error message (although if that was the case I'd expect it to freak out when loading the library, not when calling a function).  Again, someone which actually has access to Windows and tools like Dependency Walker will have to debug this, don't ask me.

File: Project.toml [hunk 1]
```diff
@@ -10,7 +10,7 @@ RDKit_jll = "03d1d220-30e6-562a-9e1a-3062d7b56d75"
 [compat]
 julia = "^1.6"
 JSON = "0.21.1"
-RDKit_jll = "2021.9.2, 2022.3.1"
+RDKit_jll = "2022.9.1"
```
Comment (giordano) [inline Project.toml:13]: Note that `2022.3.1` already allows installing version `2022.9.1`, as they're compatible per [semver](https://semver.org/) specification (except using dates as version number is all but semantic, but the package manager doesn't know that)

**Comment by giordano** [comment]
I made a debug build of RDKit for x86_64-linux-gnu: https://github.com/giordano/RDKit_jll.jl.  You can run gdb with it, should be more useful for debugging.  Running the tests with that one I get:
```
     Testing Running tests...
io: Test Failed at /home/mose/.julia/packages/RDKitMinimalLib/9YScd/test/runtests.jl:39
  Expression: mol.mol_size[] == 0x000000000000042a
   Evaluated: 0x000000000000038e == 0x000000000000042a
Stacktrace:
 [1] macro expansion
   @ ~/.julia/juliaup/julia-1.8.2+0.x64/share/julia/stdlib/v1.8/Test/src/Test.jl:464 [inlined]
 [2] macro expansion
   @ ~/.julia/packages/RDKitMinimalLib/9YScd/test/runtests.jl:39 [inlined]
 [3] macro expansion
   @ ~/.julia/juliaup/julia-1.8.2+0.x64/share/julia/stdlib/v1.8/Test/src/Test.jl:1357 [inlined]
 [4] top-level scope
   @ ~/.julia/packages/RDKitMinimalLib/9YScd/test/runtests.jl:38
io: Test Failed at /home/mose/.julia/packages/RDKitMinimalLib/9YScd/test/runtests.jl:41
  Expression: qmol.mol_size[] == 0x00000000000001eb
   Evaluated: 0x000000000000023f == 0x00000000000001eb
Stacktrace:
 [1] macro expansion
   @ ~/.julia/juliaup/julia-1.8.2+0.x64/share/julia/stdlib/v1.8/Test/src/Test.jl:464 [inlined]
 [2] macro expansion
   @ ~/.julia/packages/RDKitMinimalLib/9YScd/test/runtests.jl:41 [inlined]
 [3] macro expansion
   @ ~/.julia/juliaup/julia-1.8.2+0.x64/share/julia/stdlib/v1.8/Test/src/Test.jl:1357 [inlined]
 [4] top-level scope
   @ ~/.julia/packages/RDKitMinimalLib/9YScd/test/runtests.jl:38
Test Summary: | Pass  Fail  Total  Time
io            |    9     2     11  0.8s
ERROR: LoadError: Some tests did not pass: 9 passed, 2 failed, 0 errored, 0 broken.
in expression starting at /home/mose/.julia/packages/RDKitMinimalLib/9YScd/test/runtests.jl:37
ERROR: Package RDKitMinimalLib errored during testing
```
Hope this helps.

***Edit***: nevermind, I didn't notice you changed the expected result of this function in the tests, if I run the tests on this branch all tests are successful.  I suspect there is something wrong in the upstream library, which doesn't pop up in the debug build.  Fun.

**Comment by giordano** [comment]
I dug a bit in the source code of rdkit, unexported symbols on Windows turned out to be an upstream bug: https://github.com/rdkit/rdkit/issues/5709 (not that I was expecting something else).

File: Project.toml (hunk 1 above)
Comment (eloyfelix) [reply Project.toml:13]: thanks for the clarification!

**Comment by eloyfelix** [comment]
thanks again for all the comments/explanations. Specially for digging the windows part.
Tests now pass. As you pointed it was some changes they did in `set_2d_coords_aligned`. Still don't understand why it was working in mac xD

**Comment by giordano** [comment]
Calling conventions are different, something may work by accident.  As it was working in the debug build....  Well, all is good now! (waiting for the Windows bug to be solved upstream)

**Pull request merged**